src/bitmap.c

```diff
@@ -973,49 +974,21 @@ GdipCreateBitmapFromScan0 (int width, int height, int stride, PixelFormat format
 GpStatus
 GdipCreateBitmapFromGraphics (int width, int height, GpGraphics *graphics, GpBitmap **bitmap)
 {
-	GpBitmap 	*result;
-	FrameData	*frame;
-	BitmapData	*bitmap_data;
-	int 		stride;
-
-	stride = width * 4;
-	gdip_align_stride (stride);
-	
-	result = gdip_bitmap_new ();
-	if (!result)
-		return OutOfMemory;
-
-	result->image_format = MEMBMP;
-	result->cairo_format = CAIRO_FORMAT_ARGB32;
+	GpStatus status;
+	GpBitmap *result;
 
-	frame = gdip_frame_add(result, &gdip_image_frameDimension_page_guid);
-	if (frame == NULL) {
-		goto fail;
-	}
+	if (!graphics || !bitmap)
+		return InvalidParameter;
 
-	bitmap_data = gdip_frame_add_bitmapdata(frame);
-	if (bitmap_data == NULL) {
-		goto fail;
-	}
+	status = GdipCreateBitmapFromScan0 (width, height, 0, PixelFormat32bppPARGB, NULL, &result);
+	if (status != Ok)
+		return status;
```

Comment:
Shouldn't it use the pixel format of the passed `graphics`?

Reply:
I don't think so according to my testing. It always creates a PixelFormat32PARGB image

Reply:
Okay, I wasn't sure. Wine tests also imply PixelFormat32PARGB is used.